pkg/components/imguploader/gcsuploader.go

```diff
@@ -65,76 +56,46 @@ func (u *GCSUploader) Upload(ctx context.Context, imageDiskPath string) (string,
 	fileName += pngExt
 	key := path.Join(u.path, fileName)
 
-	var client *http.Client
-
+	var client *storage.Client
 	if u.keyFile != "" {
 		u.log.Debug("Opening key file ", u.keyFile)
 		data, err := ioutil.ReadFile(u.keyFile)
 		if err != nil {
 			return "", err
 		}
 
-		u.log.Debug("Creating JWT conf")
-		conf, err := google.JWTConfigFromJSON(data, tokenUrl)
+		u.log.Debug("Creating Google credentials from JSON")
+		creds, err := google.CredentialsFromJSON(ctx, data)
 		if err != nil {
 			return "", err
 		}
 
-		u.log.Debug("Creating HTTP client")
-		client = conf.Client(ctx)
-	} else {
-		u.log.Debug("Key file is empty, trying to use application default credentials")
-		client, err = google.DefaultClient(ctx)
+		u.log.Debug("Creating GCS client")
+		client, err = storage.NewClient(ctx, option.WithCredentials(creds))
 		if err != nil {
 			return "", err
 		}
-	}
-
-	err = u.uploadFile(client, imageDiskPath, key)
-	if err != nil {
-		return "", err
-	}
-
-	if !u.enableSignedUrls {
-		return fmt.Sprintf("https://storage.googleapis.com/%s/%s", u.bucket, key), nil
-	}
-
-	u.log.Debug("Signing GCS URL")
-	var conf *jwt.Config
-	if u.keyFile != "" {
-		jsonKey, err := ioutil.ReadFile(u.keyFile)
-		if err != nil {
-			return "", fmt.Errorf("ioutil.ReadFile: %v", err)
-		}
-		conf, err = google.JWTConfigFromJSON(jsonKey)
-		if err != nil {
-			return "", fmt.Errorf("google.JWTConfigFromJSON: %v", err)
-		}
 	} else {
-		creds, err := google.FindDefaultCredentials(ctx, storage.ScopeReadWrite)
-		if err != nil {
-			return "", fmt.Errorf("google.FindDefaultCredentials: %v", err)
-		}
-		conf, err = google.JWTConfigFromJSON(creds.JSON)
+		u.log.Debug("Creating GCS client with default application credentials")
+		client, err = storage.NewClient(ctx)
 		if err != nil {
-			return "", fmt.Errorf("google.JWTConfigFromJSON: %v", err)
+			return "", err
 		}
 	}
-	opts := &storage.SignedURLOptions{
-		Scheme:         storage.SigningSchemeV4,
-		Method:         "GET",
-		GoogleAccessID: conf.Email,
-		PrivateKey:     conf.PrivateKey,
-		Expires:        time.Now().Add(u.signedUrlExpiration),
-	}
-	signedUrl, err := storage.SignedURL(u.bucket, key, opts)
+
+	err = u.uploadFile(ctx, client, imageDiskPath, key)
 	if err != nil {
-		return "", fmt.Errorf("storage.SignedURL: %v", err)
+		return "", err
 	}
-	return signedUrl, nil
```

Comment:
Can you please tell me why this section, i.e. support for signed URLs, is removed?

Reply:
You're right. It should not be removed since this is the new feature after I modify it. I don't notice when I rebase. I will add this back.